src/main/java/xyz/wagyourtail/jsmacros/client/movement/MovementDummy.java

```diff
@@ -0,0 +1,161 @@
+package xyz.wagyourtail.jsmacros.client.movement;
+
+import net.minecraft.block.Blocks;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.entity.EntityPose;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.effect.StatusEffects;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.Arm;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MovementDummy extends LivingEntity {
+
+    public List<Vec3d> coordsHistory = new ArrayList<>();
+    private PlayerInput currentInput;
+    private int jumpingCooldown;
+
+    public MovementDummy(ClientPlayerEntity player) {
+        this(player.getEntityWorld(), player.getPos(), player.getVelocity(), player.getBoundingBox(), player.isOnGround(), player.isSprinting(), player.isSneaking());
+    }
+
+    public MovementDummy(World world, Vec3d pos, Vec3d velocity, Box hitBox, boolean onGround, boolean isSprinting, boolean isSneaking) {
```

Comment:
would the amount of code be able to get slimmed down by extending something later like `ClientPlayerEntity`?

Reply:
hmmmm, not really, since travel is the main movement method, and the other stuff I do if mostly done in the tick methods, and in those a lot of useless stuff happens.
And I like it to be able to change trim down most of the code I don't need.